src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1289,4 +1290,1373 @@ void EIRPlantLoopHeatPump::oneTimeInit(EnergyPlusData &state)
+                //        // but excluding it from coverage
+                //        ShowSevereError(state,                                                                  // LCOV_EXCL_LINE
+                //                        "EIR PLHP: Reference COP not entered and could not get default value"); // LCOV_EXCL_LINE
+                //        errorsFound = true;                                                                     // LCOV_EXCL_LINE
+                //    } else {
+                //        thisPLHP.referenceCOP = defaultVal;
+                //    }
+                //}
+
+                // A9 outdoor_air_temperature_curve_input_variable
+                std::string oaTempCurveInputVar =
+                    UtilityRoutines::MakeUPPERCase(fields.at("outdoor_air_temperature_curve_input_variable").get<std::string>());
+                thisPLHP.oaTempCurveInputVar = static_cast<OATempCurveVar>(getEnumerationValue(OATempCurveVarNamesUC, oaTempCurveInputVar));
+                if (thisPLHP.oaTempCurveInputVar == OATempCurveVar::Invalid) {
+                    thisPLHP.oaTempCurveInputVar = OATempCurveVar::DryBulb; // set to default
+                    // 2022-08-08: better give a warning for resetting to default
+                }
+
+                // A10 water_temperature_curve_input_variable
+                std::string waterTempCurveInputVar =
+                    UtilityRoutines::MakeUPPERCase(fields.at("water_temperature_curve_input_variable").get<std::string>());
+                thisPLHP.waterTempCurveInputVar =
+                    static_cast<WaterTempCurveVar>(getEnumerationValue(WaterTempCurveVarNamesUC, waterTempCurveInputVar));
+                if (thisPLHP.waterTempCurveInputVar == WaterTempCurveVar::Invalid) {
+                    thisPLHP.waterTempCurveInputVar = WaterTempCurveVar::EnteringCondenser;
+                    // 2022-08-08: give warning for resetting
+                }
+
+                // A11 normalized_capacity_function_of_temperature_curve_name
+                auto &capFtName = fields.at("normalized_capacity_function_of_temperature_curve_name");
+                thisPLHP.capFuncTempCurveIndex = Curve::GetCurveIndex(state, UtilityRoutines::MakeUPPERCase(capFtName.get<std::string>()));
+                if (thisPLHP.capFuncTempCurveIndex == 0) {
+                    ShowSevereError(state,
+                                    fmt::format("Invalid curve name for EIR PLFFHP (name={}; entered curve name: {}",
+                                                thisPLHP.name,
+                                                capFtName.get<std::string>()));
+                    errorsFound = true;
+                }
+
+                // A12 fuel_energy_input_ratio_function_of_temperature_curve_name
+                auto &eirFtName = fields.at("fuel_energy_input_ratio_function_of_temperature_curve_name");
```

Comment:
Iterators don't have to be references, they can be just plain `auto`.

Reply:
Addressed in earlier changes.